packages/jsii-pacmak/lib/targets/kotlin.ts

```diff
@@ -0,0 +1,66 @@
+import spec = require('jsii-spec');
+import { PackageInfo, Target } from '../target';
+import { shell } from '../util';
+import { KotlinGenerator } from './kotlin/kotlingenerator';
+import { getJvmPackageInfos, findJavaRuntimeLocalRepository } from './jvm/mavenutils';
+
+export class Kotlin extends Target {
+  public static toPackageInfos(assm: spec.Assembly): { [language: string]: PackageInfo } {
+    const kotlinTarget = assm.targets && assm.targets.kotlin;
+    const groupId = kotlinTarget && kotlinTarget.maven && kotlinTarget.maven.groupId;
+    const artifactId = kotlinTarget && kotlinTarget.maven && kotlinTarget.maven.artifactId;
+    const version = assm.version;
+    return getJvmPackageInfos('kotlin', groupId, artifactId, version);
+  }
+
+  public static toNativeReference(type: spec.Type, options: any) {
+    const [, ...name] = type.fqn.split('.');
+    return { java: `import ${[options.package, ...name].join('.')}` };
+  }
+
+  protected readonly generator = new KotlinGenerator();
+
+  public async build(sourceDir: string, outDir: string): Promise<void> {
+    const url = `file://${outDir}`;
+    const gradleArguments = new Array<string>();
+    for (const arg of Object.keys(this.arguments)) {
+      if (!arg.startsWith('gradle-')) { continue; }
+      gradleArguments.push(`--${arg.slice(4)}`);
```

Comment:
I think the `.slice(4)` needs to be `.slice(7)` -- I think that was copied from the Java generator where the stripped prefix is `mvn-`, and here you're stripping `gradle-` instead.